src/bindings/haskell/test/Elektra.hs

```diff
@@ -0,0 +1,63 @@
+import Elektra.Key
+import Elektra.KeySet
+import Elektra.KDB
+import Test.Hspec
+import Control.Monad
+
+main :: IO ()
+main = hspec $ do
+  describe "Key" $ do
+    it "does what i want" $ testSingleKeyOp "/test/haskell" keyGetNameSize 14
+    it "returns the correct name size" $ testSingleKeyOp "/test/haskell" keyGetNameSize 14
+    it "returns the correct full name size" $ testSingleKeyOp "/test/haskell" keyGetFullNameSize 14
+    it "returns the correct base name size" $ testSingleKeyOp "/test/haskell" keyGetBaseNameSize 8
+    it "returns the correct unescaped name size" $ testSingleKeyOp "/test/haskell" keyGetUnescapedNameSize 14
+    it "returns the correct base size" $ testSingleKeyOp "/test/haskell" keyBaseName "haskell"
+    it "returns the correct namespace" $ testSingleKeyOp "/test/haskell" keyGetNamespace KeyNsCascading
+    it "creates a new key successfully with the correct name" $ testSingleKeyOp name keyName name
+    it "sets the keys name" $ testKeyModOp name (`keySetName` otherName) keyName otherName
+    it "sets the base name" $ testKeyModOp name (`keySetBaseName` other) keyName "/test/other"
+    it "adds an escaped name" $ testKeyModOp name (`keyAddName` otherName) keyName (name ++ otherName)
+    it "adds a base name" $ testKeyModOp name (`keyAddBaseName` "haskell") keyName (name ++ "/haskell")
+    it "supports multiple operations on the same key" $ do
+      key <- keyNew name
+      keyAddBaseName key "haskell"
+      keyName key >>= (`shouldBe` name ++ "/haskell")
+      keyAddBaseName key "other"
+      keyName key >>= (`shouldBe` name ++ "/haskell/other")
+  describe "KeySet" $ do
+    it "creates a new empty keyset" $ testSingleKeySetOp 5 ksGetSize 0
+    it "creates a duplicate keyset which is not the same" $ do
+      key <- keyNew name
+      ks <- ksNew 1
+      ksAppendKey ks key
+      ks2 <- ksDup ks
+      join $ liftM2 shouldBe (ksGetSize ks) (ksGetSize ks2)
+      ksPop ks
+      join $ liftM2 shouldNotBe (ksGetSize ks) (ksGetSize ks2)
+  describe "KDB" $ it "creates a new kdb connection, saves a key, closes the connection, reopens it and checks the stored key" $ do
+      parent <- keyNew "/parent"
+      ks <- ksNew 1
+      kdb <- kdbOpen parent
+      keyNew haskellPersisted >>= ksAppendKey ks
+      kdbGet kdb ks parent
+      kdbSet kdb ks parent
```

Comment:
Does this only write to `/tests/<testname>`? Please also tag the KDB test with kdbtests, see doc/TESTING.md.